src/include/platform/internal/GenericPlatformManagerImpl_CMSISOS.ipp

```diff
@@ -0,0 +1,400 @@
+        {
+            ChipLogError(DeviceLayer, "Failed to create CHIP stack lock");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+
+    if (mChipEventQueue == nullptr)
+    {
+        mChipEventQueue = osMessageQueueNew(CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE, sizeof(ChipDeviceEvent), mEventQueueAttrPtr);
+        if (mChipEventQueue == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to allocate CHIP main event queue");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+    else
+    {
+        // Clear out any events that might be stuck in the queue, so we start
+        // with a clean slate, as if we had just re-created the queue.
+        osMessageQueueReset(mChipEventQueue);
+    }
+
+    mShouldRunEventLoop.store(false);
+
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    if (mBackgroundEventQueue == nullptr)
+    {
+        mBackgroundEventQueue =
+            osMessageQueueNew(CHIP_DEVICE_CONFIG_BG_MAX_EVENT_QUEUE_SIZE, sizeof(ChipDeviceEvent), mBgQueueAttrPtr);
+        if (mBackgroundEventQueue == nullptr)
+        {
+            ChipLogError(DeviceLayer, "Failed to allocate CHIP background event queue");
+            ExitNow(err = CHIP_ERROR_NO_MEMORY);
+        }
+    }
+    else
+    {
+        osMessageQueueReset(mBackgroundEventQueue);
+    }
+
+    mShouldRunBackgroundEventLoop.store(false);
+#endif
+
+    // Call up to the base class _InitChipStack() to perform the bulk of the initialization.
+    err = GenericPlatformManagerImpl<ImplClass>::_InitChipStack();
+    SuccessOrExit(err);
+
+exit:
+    return err;
+}
+
+template <class ImplClass>
+void GenericPlatformManagerImpl_CMSISOS<ImplClass>::_LockChipStack(void)
+{
+    osMutexAcquire(mChipStackLock, osWaitForever);
+}
+
+template <class ImplClass>
+bool GenericPlatformManagerImpl_CMSISOS<ImplClass>::_TryLockChipStack(void)
+{
+    return osMutexAcquire(mChipStackLock, 0) == osOK;
+}
+
+template <class ImplClass>
+void GenericPlatformManagerImpl_CMSISOS<ImplClass>::_UnlockChipStack(void)
+{
+    osMutexRelease(mChipStackLock);
+}
+
+#if CHIP_STACK_LOCK_TRACKING_ENABLED
+template <class ImplClass>
+bool GenericPlatformManagerImpl_CMSISOS<ImplClass>::_IsChipStackLockedByCurrentThread() const
+{
+    // If we have not started our event loop yet, return true because in that
+    // case we can't be racing against the (not yet started) event loop.
+    //
+    // Similarly, if mChipStackLock has not been created yet, might as well
+    // return true.
+    return (mEventLoopTask == nullptr) || (mChipStackLock == nullptr) || (osMutexGetOwner(mChipStackLock) == osThreadGetId());
+}
+#endif // CHIP_STACK_LOCK_TRACKING_ENABLED
+
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_CMSISOS<ImplClass>::_PostEvent(const ChipDeviceEvent * event)
+{
+    VerifyOrReturnError(mChipEventQueue != nullptr, CHIP_ERROR_INTERNAL);
```

Comment:
should this return the not-init error code?

Reply:
It could, It was returning this error code in the freeRTOS version so I didn't change it